command/agent.go

```diff
@@ -186,32 +188,42 @@ func (c *AgentCommand) Run(args []string) int {
 	}
 
 	// Validation
-	if len(c.flagConfigs) != 1 {
-		c.UI.Error("Must specify exactly one config path using -config")
+	if len(c.flagConfigs) < 1 {
+		c.UI.Error("Must specify exactly at least one config path using -config")
 		return 1
 	}
 
-	// Load the configuration file
-	config, err := agentConfig.LoadConfig(c.flagConfigs[0])
-	if err != nil {
-		c.UI.Error(fmt.Sprintf("Error loading configuration from %s: %s", c.flagConfigs[0], err))
-		return 1
+	config := agentConfig.NewConfig()
+
+	for _, configPath := range c.flagConfigs {
+		configFromPath, err := agentConfig.LoadConfig(configPath)
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("Error loading configuration from %s: %s", configPath, err))
+			return 1
+		}
+		config = config.Merge(configFromPath)
 	}
 
 	// Ensure at least one config was found.
-	if config == nil {
+	if config == agentConfig.NewConfig() {
 		c.UI.Output(wrapAtLength(
 			"No configuration read. Please provide the configuration with the " +
 				"-config flag."))
 		return 1
 	}
 
-	if config.AutoAuth == nil && config.Cache == nil {
-		c.UI.Error("No auto_auth or cache block found in config file")
+	err := config.ValidateConfig()
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error loading configuration: %s", err))
+		return 1
+	}
+
+	if config.AutoAuth == nil && config.Cache == nil && len(config.Listeners) == 0 {
```

Comment:
Shouldn't this be part of ValidateConfig?

Reply:
You're right, it should! Updated.